.github/workflows/functional-tests.yml

```diff
@@ -0,0 +1,54 @@
+---
+name: functional-tests
+
+on:
+  workflow_dispatch: ~
+  schedule:
+    - cron: '0 3 * * 1-5'
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        environment:
+          - 'qa'
+          - 'pro'
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'functionaltests/go.mod'
+
+      - uses: elastic/oblt-actions/google/auth@v1
+
+      - uses: google-github-actions/get-secretmanager-secrets@e5bb06c2ca53b244f978d33348d18317a7f263ce # v2.2.2
+        with:
+          export_to_environment: true
+          secrets: |-
+            EC_API_KEY:elastic-observability/elastic-cloud-observability-team-${{ matrix.environment }}-api-key
+
+      - run: cd functionaltests && go test -v ./
```

Comment:
It failed with

```
go: downloading github.com/apparentlymart/go-textseg/v15 v15.0.0
# github.com/elastic/apm-server/functionaltests
FAIL	github.com/elastic/apm-server/functionaltests [setup failed]
Error: main_test.go:26:2: github.com/elastic/apm-perf@v0.0.0-202412051[33](https://github.com/elastic/apm-server/actions/runs/12315688177/job/34374403286#step:6:34)854-c5983a7ff908: replacement directory ../../apm-perf does not exist
FAIL
```

See https://github.com/elastic/apm-server/actions/runs/12315688177/job/34374403286

Reply:
@v1v thanks for jumping on this this quickly! The test requires https://github.com/elastic/apm-perf/pull/197, I updated it in [320535a](https://github.com/elastic/apm-server/pull/14935/commits/320535a7071d7079b18cd413fec9614d83349220) to include the dependency from the PR. I'll trigger the workflow again.

Reply:
@v1v how can I run the pipeline? I see it has a `workflow_dispatch` trigger but I don't seem to be able to run it from the  UI?

Reply:
Good question, I created a test branch:
- https://github.com/elastic/apm-server/tree/test/funcionaltests

and the workflow is now available:
- https://github.com/elastic/apm-server/actions/workflows/functional-tests.yml

`workflow_dispatch` will be honoured as soon as the workflows exists in `main`, meanwhile, you can use the test branch for testing your things, or if you prefer, you can modify this PR and add the push event for the `branch: funcitonaltests`

Reply:
I've just synced the test branch:
- https://github.com/elastic/apm-server/actions/runs/12318045129

Reply:
Thank you!